libs/blocks/merch-card/offer-selection.js

```diff
@@ -0,0 +1,48 @@
+import { createTag } from '../../utils/utils.js';
+import { decorateButtons } from '../../utils/decorate.js';
+import '../../deps/merch-offers.js';
+
+function createDynamicSlots(el, bodySlot) {
+  const price = createTag('h5', { style: 'margin-top: 8px; margin-bottom: 16px;' });
+  price.append(createTag('span', { slot: 'price', is: 'inline-price' }));
+  bodySlot.append(price);
+
+  const p = createTag('p', { class: 'action-area' });
+  p.append(createTag('a', { slot: 'cta', is: 'checkout-link' }));
+  const footer = el.querySelector('div[slot="footer"]');
+  footer.append(p);
+  bodySlot.querySelector('p')?.setAttribute('slot', 'description');
+}
+function createMerchOffer(option) {
+  const merchOffer = createTag('merch-offer', { text: option.childNodes[0].textContent.trim() });
+  [...option.querySelector('ul').children].forEach((li, index) => {
+    const override = li.childNodes[0];
+    if (override.nodeName === '#text') {
+      if (index === 0) {
+        merchOffer.setAttribute('badge-text', override.textContent.trim());
+      } else {
+        const desc = createTag('p', { slot: 'description' });
+        desc.textContent = override.textContent.trim();
+        merchOffer.append(desc);
+      }
+    } else {
+      merchOffer.append(override);
+    }
+  });
+  decorateButtons(merchOffer);
+  return merchOffer;
+}
+
+export const initOfferSelection = (merchCard, offerSelection) => {
+  const bodySlot = merchCard.querySelector('div[slot="body-xs"]');
```

Comment:
I see this element is very particular. If the element is not found, `createDynamicSlots` will throw errors since you're appending some elements. Maybe you want to handle this particular use-case. If the element is always there you can ignore the message :)

Reply:
I will check for its presence, to make the code more stable, thx @narcis-radu .
this element is indeed always there, but nobody knows what will happen in future changes, so it's a good point.